image_search/tests/image_search_test.py

```diff
@@ -0,0 +1,144 @@
+    test_get_images: checks if the images path and images are not empty
+    test_search_image_by_text: checks if the respose is not empty
+    test_search_image_by_image: checks if the respose is not empty
+
+Usage:
+    python tests/image_search_test.py
+
+Returns:
+    0 if all tests pass, or a positive integer representing the number of failed tests.
+"""
+
+import logging
+import os
+import sys
+import time
+import unittest
+from typing import Any, Dict, List, Tuple, Type
+
+# Setup logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+kit_dir = os.path.abspath(os.path.join(current_dir, '..'))
+repo_dir = os.path.abspath(os.path.join(kit_dir, '..'))
+
+sys.path.append(kit_dir)
+sys.path.append(repo_dir)
+
+from image_search.src.image_search import ImageSearch
+
+IMAGE_TEST_DATA_PATH = os.path.join(kit_dir, 'tests/data/')
+
+
+class ImageSearchTestCase(unittest.TestCase):
+    time_start: float
+    image_search: ImageSearch
+    image_paths: List[str]
+    images: List[float]
+
+    @classmethod
+    def setUpClass(cls: Type['ImageSearchTestCase']) -> None:
+        cls.time_start = time.time()
+        cls.image_search = ImageSearch()
+        cls.image_paths, cls.images = cls.get_images()
+        cls.init_collection()
+        cls.add_images()
+
+    @classmethod
+    def get_images(cls: Type['ImageSearchTestCase']) -> Tuple[List[str], Any]:
+        paths, images = cls.image_search.get_images(IMAGE_TEST_DATA_PATH)
+        return paths, images
+
+    @classmethod
+    def init_collection(cls: Type['ImageSearchTestCase']) -> None:
+        cls.image_search.init_collection()
+
+    @classmethod
+    def add_images(cls: Type['ImageSearchTestCase']) -> None:
+        cls.image_search.add_images(IMAGE_TEST_DATA_PATH)
+
+    # Add assertions
+    def test_image_search_creation(self) -> None:
+        self.assertIsNotNone(self.image_search, 'Image Search could not be created')
+        self.assertIsNotNone(self.image_search.client, 'Vector store could not be created')
+        self.assertIsNotNone(self.image_search.embedding_function, 'Embeddings could not be created')
+        self.assertIsNotNone(self.image_search.collection, 'Collection could not be created')
+
+    def test_get_images(self) -> None:
+        self.assertGreaterEqual(len(self.image_paths), 1, 'There should be at least one path')
+        self.assertGreaterEqual(len(self.images), 1, 'There should be at least one image')
+
+    def test_search_image_by_text(self) -> None:
+        user_question = 'Show me a graph'
+        uris, distances = self.image_search.search_image_by_text(user_question)
+
+        self.assertGreaterEqual(len(uris), 1, 'There should be at least one source chunk')
+        self.assertGreaterEqual(len(distances), 1, 'There should be at least one source chunk')
```

Comment:
should we use uris and distances instead of source chunk in the error msg? same for test_search_image_by_image